### PR TITLE
Fixes #27190 - Add DSL for registering plugin fields

### PR DESCRIPTION
**Pull request by xprazak2**
```ruby
# engine.rb
Foreman::Plugin.register :foreman_openscap do
  register_graphql_query_field :tailoring_file, ::Types::TailoringFile, :record_field
end
```

### CODE REVIEW

File: app/graphql/types/query.rb [hunk 1]
```diff
@@ -12,6 +12,9 @@ def collection_field(name, type)
       end
     end
 
+    include ::Foreman::Plugin::GraphqlPluginFields
```
Comment (timogoebel) [inline app/graphql/types/query.rb:15]: Is this something we want to have in `app/graphql/types/base_object.rb` so it's available for every query or is this specific to the query field?

**Review by timogoebel — COMMENTED** [review]
Thanks, I need some more explanation to get the use case. ;-)

Also don't forget to update the docs.

**Comment by xprazak2** [comment]
I added the docs. The main aim is not to use concern in `engine.rb` to add fields from a plugin to the query like this:

```ruby
config.to_prepare do
  Types::Query.send(:include, Types::ForemanPlugin::QueryExtensions)
end
```
because it does not work well when plugin tables do not yet exist. With this, [rake errors](https://ci.theforeman.org/job/foreman_discovery-pull-request/8/database=postgresql,label=fast,ruby=2.6/console) on Jenkins should go away.

File: app/graphql/types/query.rb (hunk 1 above)
Comment (timogoebel) [reply app/graphql/types/query.rb:15]: Does this also apply for migrations?

Comment (xprazak2) [reply app/graphql/types/query.rb:15]: Yes, applies for mutations as well, updated.

**Review by timogoebel — APPROVED** [review]

File: test/unit/foreman/plugin/graphql_types_registry_test.rb [hunk 2]
```diff
@@ -51,4 +54,33 @@ def foo
...
+      err = assert_raises RuntimeError do
+        registry.register_plugin_field :woof, TestType, :custom_field
+      end
+      assert_equal err.message, "expected :record_field or :collection_field as a field_type, got custom_field"
+    end
+
+    it 'registeres plugin mutation fields' do
+      registry.register_plugin_mutation_field :boom, TestType
+      assert_equal 1, registry.plugin_mutation_fields.size
+      assert_equal :boom, registry.plugin_mutations_fields.first[:field_name]
```
Comment (timogoebel) [inline test/unit/foreman/plugin/graphql_types_registry_test.rb]:
```suggestion
      assert_equal :boom, registry. plugin_mutation_fields.first[:field_name]
```

**Review by timogoebel — CHANGES_REQUESTED** [review]
@xprazak2: Looks great, one test is failing, though.

**Comment by xprazak2** [comment]
[test foreman]

**Comment by timogoebel** [comment]
@xprazak2: Please ping me, once this is ready. I want to see this merged. :-)

**Comment by timogoebel** [comment]
[test foreman]

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Thanks, @xprazak2.

**Pull request merged**